rules/no-zero-fractions.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const MSG_ZERO_FRACTION = 'Don\'t use a zero fraction in the number.';
+const MSG_DANGLING_DOT = 'Don\'t use a dangling dot in the number.';
+
+// Groups:
+// 1 - integer part
+// 2 - dangling dot or dot with zeroes
+// 3 - dot with digits except last zeroes
+// 4 - scientific notation
+const RE_DANGLINGDOT_OR_ZERO_FRACTIONS = /^([+-]?\d*)(?:(\.0*)|(\.\d*[1-9])0+)(e[+-]?\d+)?$/;
```

Comment:
Can you add a TODO comment about using named capture groups when we target Node.js 10?

Reply:
Considering this is high performance code, I would suggest not doing this just yet as named capture groups require 1 more object allocation than numbered which introduces non-trivial slowdown.
Microbench(and that is expected to be much more allocator-friendly than real application is almost 3 times slower: https://jsperf.com/named-capture-groups

Reply:
That's not really relevant. We're not switching yet. It's going to be a long time until we can target Node.js 10, and by that time V8 will be much more optimized.

Reply:
As far as I understand, the interface is standardized already, and it requires 2 objects(unless object will reference itself, but enforcing compatibility with older versions might be a problem then, and optimizations aren't usually the goal of JS crowd anyway, so it would be a big surprise it gets implemented this way anywhere), and additional allocation will always slow down things a lot, so it's safe to say it might get faster but won't ever be up to speed with numbered groups.

Reply:
New features are always slower when first added, however, the V8 team has a good track record of optimizing features over time.

Reply:
One can't fight standard tho, and standard requires 2 objects. Memory allocation is among most important programming problems for 40 years already, and it's hard to expect any breakthroughs here.

Also, on a second thought I don't think it's even possible to use self-referencing object, as standard seem to implicit both numbered and named groups should be enumerable.